tests/components/mqtt/test_config_flow.py

```diff
@@ -1294,8 +1311,36 @@ async def test_setup_with_advanced_settings(
     assert result["data_schema"].schema[mqtt.CONF_PROTOCOL]
     assert result["data_schema"].schema[mqtt.CONF_CLIENT_CERT]
     assert result["data_schema"].schema[mqtt.CONF_CLIENT_KEY]
+    assert result["data_schema"].schema[mqtt.CONF_TRANSPORT]
+    assert result["data_schema"].schema[mqtt.CONF_WS_PATH]
+    assert result["data_schema"].schema[mqtt.CONF_WS_HEADERS]
+
+    # third iteration, advanced settings with client cert and key set and bad json payload
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            mqtt.CONF_BROKER: "test-broker",
+            mqtt.CONF_PORT: 2345,
+            mqtt.CONF_USERNAME: "user",
+            mqtt.CONF_PASSWORD: "secret",
+            mqtt.CONF_KEEPALIVE: 30,
+            "set_ca_cert": "auto",
+            "set_client_cert": True,
```

Comment:
These seem to be available as consts in `config_flow`.

Reply:
I don't think we should import from `config_flow` here.